src/ripple/app/ledger/Ledger.cpp

```diff
@@ -599,6 +605,119 @@ Ledger::peek(Keylet const& k) const
     return sle;
 }
 
+hash_set<PublicKey>
+Ledger::nUnl() const
+{
+    hash_set<PublicKey> nUnl;
+    if (auto sle = read(keylet::negativeUNL()))
+    {
+        if (sle->isFieldPresent(sfNegativeUNL))
+        {
+            auto const& nUnlData = sle->getFieldArray(sfNegativeUNL);
+            for (auto const& n : nUnlData)
+            {
+                if (n.isFieldPresent(sfPublicKey))
+                {
+                    auto d = n.getFieldVL(sfPublicKey);
+                    auto s = makeSlice(d);
+                    if (!publicKeyType(s))
+                    {
+                        continue;
+                    }
+                    nUnl.emplace(s);
+                }
+            }
+        }
+    }
+
+    return nUnl;
+}
+
+boost::optional<PublicKey>
+Ledger::nUnlToDisable() const
+{
+    if (auto sle = read(keylet::negativeUNL()))
+    {
+        if (sle->isFieldPresent(sfNegativeUNLToDisable))
+        {
+            auto d = sle->getFieldVL(sfNegativeUNLToDisable);
+            auto s = makeSlice(d);
+            if (publicKeyType(s))
+                return PublicKey(s);
+        }
+    }
+
+    return boost::none;
+}
+
+boost::optional<PublicKey>
+Ledger::nUnlToReEnable() const
+{
+    if (auto sle = read(keylet::negativeUNL()))
+    {
+        if (sle->isFieldPresent(sfNegativeUNLToReEnable))
+        {
+            auto d = sle->getFieldVL(sfNegativeUNLToReEnable);
+            auto s = makeSlice(d);
+            if (publicKeyType(s))
+                return PublicKey(s);
+        }
+    }
+
+    return boost::none;
+}
+
+void
+Ledger::updateNegativeUNL()
+{
+    if (info_.seq % 256 == 0)
```

Comment:
Consider using early returns to reduce nesting, something like:
```
void
Ledger::updateNegativeUNL()
{
    if (info_.seq % 256 != 0)
        return;

    auto sle = peek(keylet::negativeUNL());
    if (!sle)
        return;

    bool const hasToDisable = sle->isFieldPresent(sfNegativeUNLToDisable);
    bool const hasToReEnable = sle->isFieldPresent(sfNegativeUNLToReEnable);

    if (!hasToDisable && !hasToReEnable)
        return;
```

Reply:
Thanks, changed.